builtin/providers/aws/resource_aws_elb.go

```diff
@@ -725,44 +725,61 @@ func resourceAwsElbUpdate(d *schema.ResourceData, meta interface{}) error {
 
 		removed := expandStringList(os.Difference(ns).List())
 		added := expandStringList(ns.Difference(os).List())
+		subnetconn := meta.(*AWSClient).ec2conn
 
-		if len(removed) > 0 {
-			detachOpts := &elb.DetachLoadBalancerFromSubnetsInput{
-				LoadBalancerName: aws.String(d.Id()),
-				Subnets:          removed,
+		// This section sorts the removed/added subnets based on availability zones
+		dup_az_count := 0
+		for i, r_subnet_id := range removed {
+			r_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
+				SubnetIds: []*string{r_subnet_id},
+			})
+			if err != nil {
+				return err
+			}
+			for j, a_subnet_id := range added {
+				a_subnet, err := subnetconn.DescribeSubnets(&ec2.DescribeSubnetsInput{
```

Comment:
These for loops look like they just read the info, right? All the modifications seem to happen further down. I'm just suggesting moving the describe requests outside the loops, and referencing them in memory in the loops, so we don't end up describing the same subnet multiple times, if that makes sense.